`python312Packages.wandb` build failure on x86_64-linux as of `b16a62cd`

## Issue description
Build of `python312Packages.wandb` failed on x86_64-linux as of b16a62cd6010999a77f2e9ee290397278030f41f. This is currently breaking `python3Packages.wandb`.

```
error: builder for '/nix/store/h19aav078ks3dw8nwf0k9bk6b5jvvmzp-python3.12-wandb-0.18.5.drv' failed with exit code 1;
       last 10 log lines:
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED tests/unit_tests/test_mode_disabled.py::test_mode_disabled[no_wandb_core] - AssertionError
       > FAILED tests/unit_tests/test_mode_disabled.py::test_mode_disabled[wandb_core] - AssertionError
       > FAILED tests/unit_tests/test_mode_disabled.py::test_disabled_can_pickle[no_wandb_core] - AssertionError
       > FAILED tests/unit_tests/test_mode_disabled.py::test_disabled_can_pickle[wandb_core] - AssertionError
       > FAILED tests/unit_tests/test_mode_disabled.py::test_disabled_context_manager[no_wandb_core] - AssertionError
       > FAILED tests/unit_tests/test_mode_disabled.py::test_disabled_context_manager[wandb_core] - AssertionError
       > = 6 failed, 2666 passed, 80 skipped, 10 xfailed, 40 xpassed, 9 warnings in 185.83s (0:03:05) =
       For full logs, run 'nix log /nix/store/h19aav078ks3dw8nwf0k9bk6b5jvvmzp-python3.12-wandb-0.18.5.drv'.
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/12605647350)

python3Packages.wandb maintainers cc: @samuela
python312Packages.wandb maintainers cc: @samuela
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit b16a62cd6010999a77f2e9ee290397278030f41f
2. Run `nix-build -A python312Packages.wandb`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.0-1025-azure, Ubuntu, 22.04.5 LTS (Jammy Jellyfish), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.16.1`
 - nixpkgs: `/nix/store/g4ndars6r01nzj6xp1g024gjphbsjwyl-source`
```

Contents of `~/.config/nixpkgs/config.nix`:
```
{ allowUnfree = true; cudaSupport = true; cudaCapabilities = ["8.0"]; }
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/12605647350) that created this issue.
- Internal tags: 6122366028973203043 5428671873286881650

--- a/pkgs/development/python-modules/spacy/default.nix
+++ b/pkgs/development/python-modules/spacy/default.nix
@@ -49,7 +49,9 @@ buildPythonPackage rec {
   };
 
   postPatch = ''
-    sed -i "/numpy/d" pyproject.toml
+    # unpin numpy, cannot use pythonRelaxDeps because it's in build-system
+    substituteInPlace pyproject.toml setup.cfg \
+      --replace-fail ",<2.1.0" ""
   '';
 
   build-system = [

--- a/pkgs/development/python-modules/wandb/default.nix
+++ b/pkgs/development/python-modules/wandb/default.nix
@@ -296,6 +296,11 @@ buildPythonPackage rec {
       # Error in the moviepy package:
       # TypeError: must be real number, not NoneType
       "test_video_numpy_mp4"
+
+      # AssertionError: assert not _IS_INTERNAL_PROCESS
+      "test_disabled_can_pickle"
+      "test_disabled_context_manager"
+      "test_mode_disabled"
     ]
     ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64) [
       # AssertionError: assert not copy2_mock.called

